File: Leetcode/1974.cpp
class Solution {
public:
    int dist(char a, char b) {
        if (a == b) {
            return 0;
        }
        int ans = numeric_limits<int>::max();
        int counter = 0;
        char c = a;
        while (c != b) {
            if (c == 'z') {
                c = 'a';
            } else {
                ++c;
            }
            ++counter;
        }
        ans = min(ans, counter);
        counter = 0;
        c = a;
        while (c != b) {
            if (c == 'a') {
                c = 'z';
            } else {
                --c;
            }
            ++counter;
        }
        ans = min(ans, counter);
        return ans;
    }
    int minTimeToType(string word) {
        int sum = 0;
        char ch = 'a';
        for (const auto &i : word) {
            sum += dist(ch, i) + 1;
            ch = i;
        }
        return sum;
    }
};

File: Leetcode/1974.go
func dist(a, b rune) int {
	ans := math.MaxInt32
	counter := 0
	c := a
	for c != b {
		if c == 'z' {
			c = 'a'
		} else {
			c++
		}
		counter++
	}
	if ans > counter {
		ans = counter
	}
	counter = 0
	c = a
	for c != b {
		if c == 'a' {
			c = 'z'
		} else {
			c--
		}
		counter++
	}
	if ans > counter {
		ans = counter
	}
	return ans
}
func minTimeToType(word string) int {
	sum := 0
	ch := 'a'
	for _, i := range word {
		sum += dist(ch, i) + 1
		ch = i
	}
	return sum
}